wrappers/src/object_cs.cpp

```diff
@@ -35,6 +36,20 @@ using GetNativeSchemaT = void(SchemaForMarshaling schema, void* managed_callback
 namespace realm {
 namespace binding {
     extern std::function<GetNativeSchemaT> s_get_native_schema;
+
+    REALM_FORCEINLINE KeyPathArray construct_key_path_array(const ObjectSchema& object)
+    {
+        KeyPathArray keyPathArray;
+        for (auto& prop : object.persisted_properties) {
+            bool is_primitive = (unsigned short)(prop.type & ~PropertyType::Collection) == (unsigned short)prop.type;
+            if (is_primitive) {
```

Comment:
Is this just a temporary limitation or is there more to it?

Reply:
Not sure what you're asking. Is it about the weird casting above it why we're only adding singular properties to the keypaths array? (Also I realize "primitive" here is probably misleading as it also covers links - the intent was to clarify that it's filtering out collection/computed properties)

Reply:
> the intent was to clarify that it's filtering out collection/computed properties

Of course. For a second I just forgot about the intent. Thanks.